### PR TITLE
refactor(material/checkbox): simplify structural styles

### PR DESCRIPTION
**Pull request by crisbeto**
Simplifies the styles for the checkbox to make them smaller and easier to follow.

### CODE REVIEW

File: src/material/list/_list-item-hcm-indicator.scss [hunk 1]
```diff
@@ -6,25 +6,25 @@
 // its background color. Since that doesn't work in HCM, this mixin provides an alternative by
 // rendering a circle.
 @mixin private-high-contrast-list-item-indicator() {
```
Comment (crisbeto) [inline src/material/list/_list-item-hcm-indicator.scss:8]: These styles are the same, I just noticed that they weren't indented properly.

File: src/material/checkbox/_checkbox-common.scss [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+$_path-length: 29.7833385;
+$_transition-duration: 90ms;
+$_icon-size: 18px;
+$_mark-stroke-size: math.div(2, 15) * $_icon-size;
+$_indeterminate-checked-curve: cubic-bezier(0.14, 0, 0, 1);
+$_indeterminate-change-duration: 500ms;
+$_enter-curve: cubic-bezier(0, 0, 0.2, 1);
+$_exit-curve: cubic-bezier(0.4, 0, 0.6, 1);
+
+// Structural styles for a checkbox. To be shared with the selection list.
```
Comment (amysorto) [inline src/material/checkbox/_checkbox-common.scss:15]: Can you file a issue for sharing structural styles with selection list and refer to it here

**Review by amysorto — APPROVED** [review]
🎉

File: src/material/checkbox/_checkbox-common.scss (hunk 2 above)
Comment (crisbeto) [reply src/material/checkbox/_checkbox-common.scss:15]: Ah, I ended up switching the list over to these styles in this PR. Will change the comment.

**Pull request merged**